typescript/src/bitcoin.ts

```diff
@@ -412,46 +406,53 @@ export interface Client {
 export function decomposeRawTransaction(
   rawTransaction: RawTransaction
 ): DecomposedRawTransaction {
-  const toHex = (bufferWriter: any) => {
+  const toHex = (bufferWriter: any): string => {
     return bufferWriter.render().toString("hex")
   }
 
-  const vectorToRaw = (elements: any[]) => {
+  const getTxInputVector = (tx: Tx): string => {
     const buffer = bufio.write()
-    buffer.writeVarint(elements.length)
-    for (const element of elements) {
-      element.toWriter(buffer)
-    }
+    buffer.writeVarint(tx.ins.length)
+    tx.ins.forEach((input) => {
+      buffer.writeHash(input.hash)
+      buffer.writeU32(input.index)
+      buffer.writeVarBytes(input.script)
```

Comment:
Then it will add a single byte(`0x00`) representing length of `0`. 
We have this case in our existing unit tests (tested indirectly) and the unit test I added specifically for `decomposeRawTransaction`  (https://github.com/keep-network/tbtc-v2/commit/38e970bd5ad75d9ca07fb78e064894ceef195910) where one input is witness and the other is non-wotness.